pandas/tests/io/test_gcs.py

```diff
@@ -43,9 +43,24 @@ def ls(self, path, **kwargs):
     return gcs_buffer
 
 
+@pytest.fixture
+def mock_pa_filesystem(monkeypatch):
+    pa_fs = pytest.importorskip("pyarrow.fs")
+
+    class MockFileSystem(pa_fs.FileSystem):
+        @staticmethod
+        def from_uri(path):
+            to_local = path.replace("gs", "file")
+            return pa_fs.LocalFileSystem.from_uri(to_local)[0], to_local
```

Comment:
Could we somehow also assert that this is being called? To actually ensure we get here, because if not (and it's still using fsspec), the tests still pass I assume. 
(eg define a list that gets appended with some value inside this function, and yield that from this fixture, so we can see that it was appended after calling read_parquet. Not sure if that's worth the complexity)

Reply:
Sure. I added a print statement and validated that the message was printed with `capsys`